### PR TITLE
Fixes #20626 - Periodic execution plan cleaner

### PR DESCRIPTION
**Pull request by adamruzicka**

### CODE REVIEW

**Comment by iNecas** [comment]
One thing that came out of the testing: the cleaner should emit some info log messages about it's activity: it can be very useful

**Comment by iNecas** [comment]
The log needs to probably go into Dynflow

File: lib/smart_proxy_dynflow_core/settings.rb [hunk 1]
```diff
@@ -40,6 +40,7 @@ class Settings < OpenStruct
         :plugins => {},
         :pid_file => '/var/run/foreman-proxy/smart_proxy_dynflow_core.pid',
         :daemonize => false,
+        :execution_plan_cleaner_age => 60 * 60 * 24,
```
Comment (iNecas) [inline lib/smart_proxy_dynflow_core/settings.rb]: You need to add this also to `PLUGIN_SETTINGS` for the code to properly load the data from settings

**Comment by iNecas** [comment]
Tested and works well, after doing some changes mentioned in the review

**Comment by adamruzicka** [comment]
Addressed comments, logging is handled in https://github.com/Dynflow/dynflow/pull/248

**Comment by iNecas** [comment]
I don't see the code that would address the comments: forgot to push?

**Comment by adamruzicka** [comment]
Pushed now...

File: settings.d/dynflow.yml.example [hunk 2]
```diff
@@ -2,3 +2,4 @@
 :enabled: true
 :database: /var/lib/foreman-proxy/dynflow/dynflow.sqlite
 :core_url: 'http://127.0.0.1:8008'
+:cleaner_max_age: 86400 # 1 day
```
Comment (iNecas) [inline settings.d/dynflow.yml.example]: Why this one?

**Comment by iNecas** [comment]
I've released dynflow 0.8.27, please bump the dependency here.

**Comment by iNecas** [comment]
Test failures unrelated. Merging

**Comment by iNecas** [comment]
Tested and works great. Thanks @adamruzicka

**Pull request merged**